test/cppsuite/test_harness/workload/workload_tracking.h

```diff
@@ -94,12 +94,69 @@ class workload_tracking : public component {
         testutil_check(_session->create(
           _session.get(), _operation_table_name.c_str(), _operation_table_config.c_str()));
         log_msg(LOG_TRACE, "Operations tracking created");
+
+        /*
+         * Open sweep cursor. This cursor will be used to clear out obsolete data from the tracking
+         * table.
+         */
+        _sweep_cursor = _session.open_scoped_cursor(_operation_table_name.c_str());
+        log_msg(LOG_TRACE, "Tracking table sweep initialized");
     }
 
     void
-    run() override final
+    do_work() override final
     {
-        /* Does not do anything. */
+        WT_DECL_RET;
+        wt_timestamp_t timestamp;
+        int64_t collection_id;
+        const char *key;
+        char *sweep_key;
+        bool globally_visible_update_found;
+
+        key = sweep_key = nullptr;
+        globally_visible_update_found = false;
+
+        /* Take a copy of the oldest so that we sweep with a consistent timestamp. */
+        wt_timestamp_t oldest_ts = _tsm.get_oldest_ts();
+
+        /* If we have a position, give it up. */
+        testutil_check(_sweep_cursor->reset(_sweep_cursor.get()));
+
+        while ((ret = _sweep_cursor->prev(_sweep_cursor.get())) == 0) {
+            testutil_check(
+              _sweep_cursor->get_key(_sweep_cursor.get(), &collection_id, &key, &timestamp));
+            /*
+             * If we're on a new key, reset the check. We want to track whether we have a globally
```

Comment:
Oh actually, I just realised I should check `collection_id` here too since we could have the same key for different tables.

Reply:
c9e69d2